### ISSUE TITLE
JAVASCRIPT TypeError thrown by use of facetBy second parameter

### ISSUE THREAD

**Issue by georgeajit** [marklogic/java-client-api#1286]
So we can address your issue, please include the following:
### Version of MarkLogic Java Client API
develop branch (post 5.3.2)

### Version of MarkLogic Server

10.0-6

### Java version

1.8 u 261

### OS and version

Windows IntelliJ IDE

### Input: Some code to illustrate the problem, preferably in a state that can be independently reproduced on our end
```
ModifyPlan plan1 = p.fromView("opticFunctionalTest", "detail")
            .orderBy(p.schemaCol("opticFunctionalTest", "detail", "id"));
    ModifyPlan plan2 = p.fromView("opticFunctionalTest", "master")
            .orderBy(p.schemaCol("opticFunctionalTest", "master", "id"));
    PlanParamExpr idParam = p.param("ID");
    ModifyPlan plan3 = plan1.joinInner(plan2)
            .where(
                    p.eq(p.schemaCol("opticFunctionalTest", "master", "id"), idParam)
            )
            .facetBy(p.colSeq(p.schemaCol("opticFunctionalTest", "detail", "color")),
                    p.col("id")
            );
            //.orderBy(p.asc(p.schemaCol("opticFunctionalTest", "detail", "id")));
    JacksonHandle jacksonHandle = new JacksonHandle();
    jacksonHandle.setMimetype("application/json");

    rowMgr.resultDoc(plan3.bindParam("ID", 1), jacksonHandle);
...
...
```

### Actual output: What did you observe? What errors did you see? Can you attach the logs? (Java logs, MarkLogic logs)
```
Local message: failed to apply resource at rows: Internal Server Error. Server Message: JS-JAVASCRIPT: aggritems = aggregates.map(aggregate => columnProp(aggregate.outputColumn())); -- Error running JavaScript request: TypeError: aggregate.outputColumn is not a function . See the MarkLogic server error log for further detail.
com.marklogic.client.FailedRequestException: Local message: failed to apply resource at rows: Internal Server Error. Server Message: JS-JAVASCRIPT: aggritems = aggregates.map(aggregate => columnProp(aggregate.outputColumn())); -- Error running JavaScript request: TypeError: aggregate.outputColumn is not a function . See the M
```
### Expected output: What specifically did you expect to happen?
Rows
### Alternatives: What else have you tried, actual/expected?
N/A

**Comment by ehennum** [comment]
Thanks for catching this issue.

The root cause here was a server bug that hid another server bug (internal numbers 55813 and 55810), both of which are fixed in 10.0-6.1

Please retest with a 10.0-6.1, 10.0-7, or trunk nightly

**Comment by georgeajit** [comment]
Will be tested when server release builds are available.
